src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -116,7 +116,11 @@ internal sealed class VisualStudioProject
         /// The workspace snapshot will only have a document with  <see cref="DynamicFileInfo.FilePath"/> (the value) but not the
         /// original dynamic file path (the key).
         /// </summary>
-        private readonly Dictionary<string, string?> _dynamicFilePathMaps = new();
+        /// <remarks>
+        /// We use the same string comparer as in the <see cref="BatchingDocumentCollection"/> used by _sourceFiles, below, as these
+        /// files are added to that collection too.
+        /// </remarks>
+        private readonly Dictionary<string, string?> _dynamicFilePathMaps = new(StringComparer.OrdinalIgnoreCase);
```

Comment:
This is only VSWin right?

Reply:
Yes, but even if its not, its probably better that source files and dynamic files make the same mistakes 😛